buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -554,7 +562,11 @@ public OpenSearchNode singleNode() {
     private void addWaitForClusterHealth() {
         waitConditions.put("cluster health yellow", (node) -> {
             try {
-                WaitForHttpResource wait = new WaitForHttpResource("http", getFirstNode().getHttpSocketURI(), nodes.size());
+                WaitForHttpResource wait = new WaitForHttpResource(
+                    getFirstNode().getHttpProtocol(),
```

Comment:
Does this mean that a user testing bwc with security will first need to ensure that

```
plugins.security.ssl.http.enabled: true
```

to ensure that the node with the security plugin installed are running with `https` and then also set:

```
http.protocol: https
```

so that this single call will use `https` when making the request? Is there potentially another way of getting this setting from the plugin to control the scheme here or is that impossible?

Reply:
I believe so. We need this setup because we are trying to make the options configurable for two different purposes. On the Security Plugin side, we are only concerned with HTTP vs. HTTPS but with core, we need to handle HTTP vs. HTTPS or as DB mentioned other things like HTTP/2 etc.